dashboard/app/controllers/level_starter_assets_controller.rb

```diff
@@ -28,20 +18,71 @@ def show
   # Returns requested file body as an IO stream.
   def file
     friendly_name = "#{params[:filename]}.#{params[:format]}"
-    guid_name = @level.starter_assets[friendly_name]
-    file_obj = get_file_object(prefix(guid_name))
-    content_type = file_content_type(File.extname(guid_name))
+    uuid_name = @level.starter_assets[friendly_name]
+    file_obj = get_object(uuid_name)
+    content_type = file_content_type(File.extname(uuid_name))
 
     expires_in 1.hour, public: true
     send_data read_file(file_obj), type: content_type, disposition: 'inline'
   end
 
+  # POST /level_starter_assets/:level_name
+  def upload
```

Comment:
I think the standard name for this would be `create`, and the standard name for `file` would be `get`.